src/pykeen/typing.py

```diff
@@ -61,21 +62,28 @@
 
 X = TypeVar("X")
 
+# some PyTorch functions to not properly propagate types (e.g., float() does not return FloatTensor but Tensor)
+# however it is still useful to distinguish float tensors from long ones
+# to make the switch easier once PyTorch improves typing, we use a global type alias inside PyKEEN.
+BoolTensor: TypeAlias = torch.Tensor  # replace by torch.BoolTensor
+FloatTensor: TypeAlias = torch.Tensor  # replace by torch.FloatTensor
+LongTensor: TypeAlias = torch.Tensor  # replace by torch.LongTensor
```

Comment:
should we define these as typevars that are bound to tensor so it knows there's a difference between them?

Reply:
This seems to make typing quite difficult, since we also use it as a return type, e.g.,
```
src/pykeen/inverse.py:43: error: A function returning TypeVar should receive at least one argument containing the same TypeVar  [type-var]
src/pykeen/inverse.py:43: note: Consider using the upper bound "Any" instead
```

Reply:
That’s really interesting. Not a blocker tho